src/main/java/org/polyfill/services/PreSortPolyfillQueryService.java

```diff
@@ -46,32 +48,37 @@ public void loadConfigs() throws IOException {
         this.polyfills = getPolyfillsMap(polyfillsDirPath);
         this.sortedPolyfills = getDependencySortedPolyfills();
         this.baselineVersions = getConfig(baselineVersionsPath);
+        this.aliases = getConfig(aliasesPath);
     }
 
     /**
-     * TODO:
-     * Filter polyfill list by user agent, then get the requested feature from
-     * polyfill list along with its dependencies.
-     * The result list is ordered by dependencies.
-     * @param userAgent user agent object
-     * @param name feature name; e.g. Array.of
-     * @return the filtered list
-     */
-    public List<Polyfill> getPolyfillByFeature(UserAgent userAgent, String name) {
-        List<Polyfill> filteredPolyfill = new ArrayList<>();
-        return filteredPolyfill;
-    }
-
-    /**
-     * TODO:
      * Filter polyfill list by alias and user agent and order it by dependencies
      * @param userAgent user agent object
-     * @param alias polyfill set alias; e.g. es6, then we return all es6 polyfills
+     * @param featureNames list of feature group alias names or feature names, delimited by ","
+     *                     e.g. "es6,es5,Array.of"
      * @return the filtered list
      */
-    public List<Polyfill> getPolyfillsByAlias(UserAgent userAgent, String alias) {
-        List<Polyfill> filteredPolyfill = new ArrayList<>();
-        return filteredPolyfill;
+    public List<Polyfill> getPolyfillsByFeatures(UserAgent userAgent, String featureNames) {
+        if (meetsBaseline(userAgent)) {
+            Set<String> featureSet = new HashSet<>();
+
+            List<String> featureNamesList = Arrays.asList(featureNames.split(","));
```

Comment:
Yea I thought about that too, but when we receive the URL, the features are in that format, that is, separated by ",". If I handle it here, we don't need to do anything else when using it. Anyways, I can change it to a list.